### DISCUSSION TITLE
How to switch to oracle mode?

### DISCUSSION THREAD

**Discussion by imranzaheer612** [IvorySQL/IvorySQL#659] [General]
In previous version we used to switch modes using 'set comaptibility_mode'.
I noticed there is something changed in 3.2 (latest versions), its 'SET ivorysql.compatible=oracle;' Now.
But cannot find this variable in neither `postgresql.conf` nor in `ivorysql.conf`
Also cannot change it by the psql
```
postgres=# show ivorysql.compatible;
2024-04-26 09:15:18.316 PKT [131010] ERROR:  unrecognized configuration parameter "ivorysql.compatible"
2024-04-26 09:15:18.316 PKT [131010] STATEMENT:  show ivorysql.compatible;
ERROR:  unrecognized configuration parameter "ivorysql.compatible"
postgres=# show compatible_mode;
2024-04-26 09:15:26.328 PKT [131010] ERROR:  unrecognized configuration parameter "compatible_mode"
2024-04-26 09:15:26.328 PKT [131010] STATEMENT:  show compatible_mode;
ERROR:  unrecognized configuration parameter "compatible_mode"
postgres=# 
```

How to use the server in oracle mode?

**Comment by gaoxueyu** [comment]
In IvorySQL3 version, if you want use oracle compatible mode, you could try as following:
step-1: initdb with oracle compatible mode
 initdb -D data -m oracle

step-2: start ivorysql
pg_ctl -D data -l logfile start

step-3: connect oracle port via 1521 and change compatible mode via "set ivorysql.compatible_mode"
[postgres@ivorysql ~]$ psql -p 1521
psql (16.2)
Type "help" for help.

postgres=# select version();
                                                        version                              
 PostgreSQL 16.2 (IvorySQL 3.2) on x86_64-pc-linux-gnu, compiled by gcc (GCC) 4.8.5 20150623 (
Red Hat 4.8.5-44), 64-bit
(1 row)

postgres=# show ivorysql.compatible_mode;
 ivorysql.compatible_mode 
 oracle
(1 row)

postgres=# set ivorysql.compatible_mode="pg";
SET
postgres=# show ivorysql.compatible_mode;
 ivorysql.compatible_mode 
 pg
(1 row)

Feel free to let me know if anything is confusing.

**Reply by imranzaheer612** [reply]
Thanks. Are these steps documented somewhere? Because when I tried looking into the [docs](https://docs.ivorysql.org/en/ivorysql-doc/v3.2/v3.2/8#compatible_mode-enum). it only showed me the old method of setting the configurations for oracle mode.

**Reply by balinorLiang** [reply]
For GUC variables, please refer to [GUC framework](https://docs.ivorysql.org/en/ivorysql-doc/v3.2/v3.2/15), there are some usage cases, like ivorysql.compatible and ivorysql.datetime_ignore_nls_mask etc. The old method you were pointed out in docs, will be modified and updated ASAP.

**Comment by imranzaheer612** [comment]
Ok Thanks guys. Closing it now.
